cardano-testnet/src/Testnet/Components/Query.hs

```diff
@@ -240,22 +243,22 @@ checkDRepsNumber sbe configurationFile socketPath execConfig expectedDRepsNb = d
   QueryTipLocalStateOutput{mEpoch} <- P.execCliStdoutToJson execConfig [ "query", "tip" ]
   currentEpoch <- H.evalMaybe mEpoch
   let terminationEpoch = succ . succ $ currentEpoch
-  void $ H.evalMaybeM $ checkDRepsNumber' sbe configurationFile socketPath terminationEpoch expectedDRepsNb
+  void $ H.evalMaybeM $ getDRepInfo sbe configurationFile socketPath terminationEpoch expectedDRepsNb
 
--- | @checkDRepsNumber' config socket terminationEpoch n@
+-- | @getDRepInfo config socket terminationEpoch n@
 -- wait until @terminationEpoch@ for the number of DReps being @n@. If
 -- this number is not attained before @terminationEpoch@, the test is failed.
 -- So if you call this function, you are expecting the number of DReps to already
 -- be @n@, or to be @n@ before @terminationEpoch@
-checkDRepsNumber' ::
+getDRepInfo ::
```

Comment:
I'd probably keep the name `checkDRepsNumber'` because the `check`-prefix indicates that this function checks something, and if this something is wrong, the test will be failed. That is why you need to supply the expected number of DReps: to check that it's what you expect. If the function didn't require this argument, `get` would be a fine prefix.

Reply:
I don't really mind the name, but I would disagree. I would argue it is more of a side-effect than the main purpose of the function. Note that there is `checkDRepsNumber` without apostrophe that does exactly what you say. This one returns the `DRepState` once the number of `DReps` matches as long as it is before the given `EpochNo`.

So calling this `checkDRepsNumber'` is a bit like instead of calling the division operation `div`  we called it `checkThatDivisorIsNotZero'` because it fails when it is.

My theory is that whoever named it didn't care about the name because it was just an auxiliar function to `checkDRepsNumber` and wasn't exported, and I thought it was my duty to give it a more meaningful name because I am exporting it.

Are you convinced? :)

Maybe we can call it `waitForDRepsAndGetState` or something like that

Reply:
@palas> please proceed in the way you want :+1: No strong opinion from my side.

Reply:
I think this function is too specialized by asking the caller to provide an expected number of dreps. We have `foldEpochState` for this purpose. It makes more sense imo to have a function that registers a number of dreps and a test that checks this function works as expected rather than checking each time we have the expected number of dreps.

Reply:
That would make it more reusable, you are right. I will revert this and do a generic function that just receives a condition